### PR TITLE
Image transforms add center crop

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Adds `center_crop` to the image transforms library, to be used in the future image processors. Performs equivalent processing as done in the `ImageFeatureExtractionMixin`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

### CODE REVIEW

File: src/transformers/image_transforms.py [hunk 1]
```diff
@@ -316,3 +316,95 @@ def normalize(
...
+    Args:
+        image (`np.ndarray`):
+            The image to crop.
+        size (`Tuple[int, int]`):
+            The target size for the cropped image.
+        data_format (`str` or `ChannelDimension`, *optional*):
+            The channel dimension format for the output image. Can be one of:
+                    - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                    - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
+            If `None`, will use the inferred format of the input image.
```
Comment (sgugger) [inline src/transformers/image_transforms.py:340]:
```suggestion
            If unset, will use the inferred format of the input image.
```

File: src/transformers/image_transforms.py [hunk 2]
```diff
@@ -316,3 +316,95 @@ def normalize(
...
+
+    Args:
+        image (`np.ndarray`):
+            The image to crop.
+        size (`Tuple[int, int]`):
+            The target size for the cropped image.
+        data_format (`str` or `ChannelDimension`, *optional*):
+            The channel dimension format for the output image. Can be one of:
+                    - `"channels_first"` or `ChannelDimension.FIRST`: image in (num_channels, height, width) format.
+                    - `"channels_last"` or `ChannelDimension.LAST`: image in (height, width, num_channels) format.
```
Comment (sgugger) [inline src/transformers/image_transforms.py:339]: We can do with one less level of indentation

File: src/transformers/image_transforms.py [hunk 3]
```diff
@@ -316,3 +316,95 @@ def normalize(
...
+            If `None`, will use the inferred format of the input image.
+        return_numpy (`bool`, *optional*):
+            Whether or not to return the cropped image as a numpy array. Used for backwards compatibility with the
+            previous ImageFeatureExtractionMixin method.
+                - `None` (default): will return the same type as the input image.
```
Comment (sgugger) [inline src/transformers/image_transforms.py:344]:
```suggestion
                - Unset: will return the same type as the input image.
```

File: src/transformers/image_transforms.py [hunk 4]
```diff
@@ -316,3 +316,95 @@ def normalize(
...
+    # We perform the crop in (C, H, W) format and then convert to the output format
+    image = to_channel_dimension_format(image, ChannelDimension.FIRST)
+
+    orig_height, orig_width = get_image_size(image)
+    crop_height, crop_width = size
+
+    top = (orig_height - crop_height) // 2
+    bottom = top + crop_height  # In case size is odd, (image_shape[0] + size[0]) // 2 won't give the proper result.
+    left = (orig_width - crop_width) // 2
+    right = left + crop_width  # In case size is odd, (image_shape[1] + size[1]) // 2 won't give the proper result.
```
Comment (sgugger) [inline src/transformers/image_transforms.py:378]: Nice comments! Nit: I prefer them on top of the line they comment.

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding this!

File: src/transformers/image_transforms.py (hunk 4 above)
Comment (amyeroberts) [reply src/transformers/image_transforms.py:378]: I can't take credit - they're your comments! I'll move them above 👍

Comment (amyeroberts) [reply src/transformers/image_transforms.py:378]: I can't take credit - they're your comments! I'll move them above 👍

Comment (sgugger) [reply src/transformers/image_transforms.py:378]: Ah ah!

**Review by NielsRogge — APPROVED** [review]
Thanks for adding!

**Review by alaradirik — APPROVED** [review]
Looks good to me!

**Pull request merged**